File: 0001-TwoSum/TwoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const seen = {};
  let indices = [];
  for (let i = 0; i < nums.length; i++) {
    const complement = target - nums[i];
    if (seen[complement] !== undefined) {
      indices = [seen[complement], i];
      break;
    } else {
      seen[nums[i]] = i;
    }
  }
  return indices;
};

File: 0001-TwoSum/TwoSum.py
class Solution:
    def twoSum(self, nums: list[int], target: int)->list[int]:
        seen: dict[int, int] = {}
        for i, num in enumerate(nums):
            complement: int = target - num
            if complement in seen:
                return [seen[complement], i]
            else:
                seen[num] = i
        return []